File: GFG POTD/032-February-2026/007-max-sum-in-the-configuration.java
// https://www.geeksforgeeks.org/problems/max-sum-in-the-configuration/1/




class Solution {
    int maxSum(int[] arr) {
        int n = arr.length;
        
        int currRotation  = 0;
        int totalSum      = 0;
        
        for (int i = 0; i < n; i ++) {
            currRotation += (i * arr[i]);
            totalSum += arr[i];
        }
        
        int maxVal = currRotation;
        for (int i = 1; i < n; i ++) {
            int nextRotation = ((currRotation + totalSum) - (n * arr[n - i]));
            maxVal = Math.max(maxVal, nextRotation);
            currRotation = nextRotation;
        }
        return maxVal;
    }
}

File: GFG POTD/032-February-2026/007-max-sum-in-the-configuration.py
# https://www.geeksforgeeks.org/problems/max-sum-in-the-configuration/1/




class Solution:
    def maxSum(self, arr): 
        n = len(arr)
        
        currRotation = totalSum = 0
        
        for i in range(n):
            currRotation += (i * arr[i])
            totalSum += arr[i]
            
        maxVal = currRotation
        
        for i in range(1, n):
            nextRotation = (currRotation + totalSum - (n * arr[n - i]))
            maxVal = max(maxVal, nextRotation)
            currRotation = nextRotation
            
        return maxVal